src/twodtree.h

```diff
@@ -0,0 +1,108 @@
+// Copyright 2025 The Manifold Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <algorithm>
+
+#include "./utils.h"
+#include "manifold/common.h"
+#include "manifold/optional_assert.h"
+#include "manifold/vec_view.h"
+
+namespace manifold {
+
+template <typename P, bool sortX = true>
```

Comment:
I'll take a wild guess that the next version of this will be 3D, splitting X, Y, Z, X, Y, Z. Maybe `bool sortX` should be `int sortAxis`?